front/src/applications/stdcmV2/components/StdcmVias.tsx

```diff
@@ -0,0 +1,104 @@
+import StdcmDefaultCard from './StdcmDefaultCard';
+import StdcmInputVia from './StdcmInputVia';
+import StdcmOperationalPoint from './StdcmOperationalPoint';
+import type { StdcmConfigCardProps } from '../types';
+
+const StdcmVias = ({ disabled = false, setCurrentSimulationInputs }: StdcmConfigCardProps) => {
+  const { t } = useTranslation('stdcm');
+  const dispatch = useAppDispatch();
+  const { getPathSteps } = useOsrdConfSelectors();
+  const { deleteViaV2, updatePathSteps, updateViaStopTimeV2 } =
+    useOsrdConfActions() as StdcmConfSliceActions;
+  const pathSteps = useSelector(getPathSteps);
+
+  const intermediatePoints = useMemo(() => pathSteps.slice(1, -1), [pathSteps]);
+
+  const updatePathStepsList = (pathStep: PathStep | null, index: number) => {
+    const newPathSteps = replaceElementAtIndex(pathSteps, index, pathStep);
+    dispatch(updatePathSteps(newPathSteps));
+  };
+
+  const updatePathStepStopTime = (stopTime: string, index: number) => {
+    const pathStepToUpdate = pathSteps[index];
+    if (!pathStepToUpdate) return;
+    dispatch(
+      updateViaStopTimeV2({
+        via: pathStepToUpdate,
+        duration: formatDurationAsISO8601(Number(stopTime) * 60),
+      })
+    );
+  };
+
+  useEffect(() => {
+    setCurrentSimulationInputs((prevState) => ({
+      ...prevState,
+      pathSteps,
+    }));
+  }, [pathSteps]);
+
+  return (
+    <div className="stdcm-v2-vias-list">
+      {intermediatePoints.length > 0 &&
+        intermediatePoints.map((pathStep, index) => {
+          const pathStepId = index + 1;
+          return (
+            <StdcmCard
+              key={pathStepId}
+              name={t('trainPath.vias')}
+              title={
+                <div className="stdcm-v2-via-icons">
+                  <span>
+                    <img src={IntermediatePointIcon} alt="intermediate-point" />
+                  </span>
+                  <button type="button" onClick={() => dispatch(deleteViaV2(index))}>
+                    {t('translation:common.delete')}
+                  </button>
+                </div>
+              }
+              hasTip
+              disabled={disabled}
+            >
+              <div className="stdcm-v2-vias">
+                <StdcmOperationalPoint
+                  updatePoint={(e) => updatePathStepsList(e, pathStepId)}
+                  point={pathStep}
+                  disabled={disabled}
+                />
+              </div>
+              {pathStep && (
+                <StdcmInputVia
+                  pathStep={pathStep}
+                  updatePathStepStopTime={(e) => updatePathStepStopTime(e, pathStepId)}
+                />
+              )}
+            </StdcmCard>
+          );
+        })}
+      <StdcmDefaultCard
+        hasTip
+        text={t('trainPath.addVia')}
+        Icon={<Location size="lg" variant="base" />}
+        onClick={() =>
+          dispatch(updatePathSteps(addElementAtIndex(pathSteps, pathSteps.length - 1, null)))
```

Comment:
Why are you adding null at the end of the array ?
```suggestion
          const newPathSteps = pathSteps.push(null);
          dispatch(updatePathSteps(newPathSteps))
```

Reply:
We need to add `null` to display a new card and define a new point, as we cannot add an empty PathStep